src/Engine/Image.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Engine;
+
+use Imagine\Filter\Basic\Crop;
+use Imagine\Image\Box;
+use Imagine\Image\Point;
+
+class Image
+{
+
+    /** @var string Original URL of image. */
+    private $imageUrl;
+
+    /** @var int[] Array of floats with keys: `x`, `y`, `width`, and `height`. */
+    private $crop;
+
+    /** @var string Image data. */
+    private $data;
+
+    /** @var int Image size in bytes. */
+    private $size;
+
+    /**
+     * @param int[] $crop
+     */
+    public function __construct(string $imageUrl, array $crop = [])
+    {
+        $this->imageUrl = $imageUrl;
+        $this->crop = $crop;
+    }
+
+    public function geturl(): string
+    {
+        return $this->imageUrl;
+    }
+
+    public function needsCropping(): bool
+    {
+        return isset($this->crop['width']) && $this->crop['width'] > 0
+            && isset($this->crop['height']) && $this->crop['height'] > 0;
+    }
+
+    public function getCrop(): Crop
+    {
+        return new Crop(
+            new Point($this->crop['x'], $this->crop['y']),
+            new Box($this->crop['width'], $this->crop['height'])
+        );
+    }
+
+    public function getData(): string
```

Comment:
In theory, we're not guaranteed to respect the signature requirements, if getData() is called without setData() being called first. I see two main options:

1 - (preferred, but unsure if it would work) have a single method that returns the "identifier" of an image. This could be an URL or image data, and the caller wouldn't know. That would avoid the needsCropping() checks, but assumes that every caller can handle the data and URL cases transparently, whereas tesseract only accepts image data.

2 - (easier) Document that setData() is required before getData(), and throw an exception from getData() if $this->data is null. There could also be an hasData() method that callers should use beforehand, instead of needsCropping().